опубликование Sprint_5

KristyConstanta opened the pull request:
modified:   README.md
	new file:   conftest.py
	new file:   locators.py
	new file:   tests/test_login.py
	new file:   tests/test_logout.py
	new file:   tests/test_move_constructor.py
	new file:   tests/test_move_personal_account.py
	new file:   tests/test_registration.py
	new file:   tests/test_switch_tabs.py

KristyConstanta committed:
new file:   .gitignore
	modified:   README.md
	new file:   conftest.py
	new file:   locators.py
	new file:   tests/test_login.py
	new file:   tests/test_logout.py
	new file:   tests/test_move_constructor.py
	new file:   tests/test_move_personal_account.py
	new file:   tests/test_registration.py
	new file:   tests/test_switch_tabs.py

DolgoarshinnykhLG commented on tests/test_login.py:
```diff
@@ -0,0 +1,65 @@
+locators = Locators()
+email = 'kristy_rassudova_31@yandex.ru'
+password = '510547'
+
```
Необходимо исправить: в модуле с тестами должны быть только тесты. Тестовые данные стоит вынести в отдельный модуль data, а класс с локаторами использовать по месту в тестах

DolgoarshinnykhLG commented on tests/test_login.py:
```diff
@@ -0,0 +1,65 @@
+password = '510547'
+
+# Вход по кнопке «Войти в аккаунт» на главной
+def test_login_button_main_page(driver):
```
Нужно исправить здесь и далее: Для корректного запуска тестов необходимо код с шагами теста поместить в тестовый метод (нейминг начинается с test_*), а метод - в тестовый класс (нейминг начинается с Test*). Необходимо привести к такому формату все модули внутри пакета tests

DolgoarshinnykhLG commented on tests/test_login.py:
```diff
@@ -0,0 +1,65 @@
+
+# Вход по кнопке «Войти в аккаунт» на главной
+def test_login_button_main_page(driver):
+    driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше здесь и далее: адреса лучше вынести во внешний модуль(например, urls). Они могут измениться, что затруднит поддержку тестов

DolgoarshinnykhLG commented on tests/test_switch_tabs.py:
```diff
@@ -0,0 +1,48 @@
+    driver.find_element(*locators.sauces_section).click()
+    driver.find_element(*locators.buns_section).click()
+    assert driver.find_element(*locators.buns_section).text == "Булки"
+
```
Можно лучше: шаги этих тестов схожи. Можно их объединить с помощью параметризации

DolgoarshinnykhLG commented on tests/test_switch_tabs.py:
```diff
@@ -0,0 +1,48 @@
+    driver.find_element(*locators.login_button).click()
+    WebDriverWait(driver, 8).until(expected_conditions.visibility_of_element_located(locators.make_an_order_button))
+    driver.find_element(*locators.fillings_section).click()
+    assert driver.find_element(*locators.fillings_section).text == "Начинки"
```
Необходимо исправить: проверка получилась хрупкой, она никак не проверяет, что активный раздел действительно изменился после перехода

DolgoarshinnykhLG commented on tests/test_switch_tabs.py:
```diff
@@ -0,0 +1,48 @@
+    WebDriverWait(driver, 8).until(expected_conditions.visibility_of_element_located(locators.make_an_order_button))
+    driver.find_element(*locators.fillings_section).click()
+    driver.find_element(*locators.sauces_section).click()
+    assert driver.find_element(*locators.sauces_section).text == "Соусы"
```
Необходимо исправить: проверка получилась хрупкой, она никак не проверяет, что активный раздел действительно изменился после перехода

DolgoarshinnykhLG commented on tests/test_switch_tabs.py:
```diff
@@ -0,0 +1,48 @@
+    WebDriverWait(driver, 8).until(expected_conditions.visibility_of_element_located(locators.make_an_order_button))
+    driver.find_element(*locators.sauces_section).click()
+    driver.find_element(*locators.buns_section).click()
+    assert driver.find_element(*locators.buns_section).text == "Булки"
```
Необходимо исправить: проверка получилась хрупкой, она никак не проверяет, что активный раздел действительно изменился после перехода

KristyConstanta committed:
new file:   data.py
	modified:   locators.py
	modified:   tests/test_login.py
	modified:   tests/test_logout.py
	modified:   tests/test_move_constructor.py
	modified:   tests/test_move_personal_account.py
	modified:   tests/test_registration.py
	modified:   tests/test_switch_tabs.py
